kernel/src/path.rs

```diff
@@ -70,11 +70,12 @@ pub(crate) struct ParsedLogPath<Location: AsUrl = FileMeta> {
 }
 
 // Internal helper used by TryFrom<FileMeta> below. It parses a fixed-length string into the numeric
-// type expected by the caller. A wrong length produces an error, even if the parse succeeded.
-fn parse_path_part<T: FromStr>(value: &str, expect_len: usize, location: &Url) -> DeltaResult<T> {
+// type expected by the caller. A parsing failure returns None. A wrong length produces None, even
+// if the parse succeeded.
+fn parse_path_part<T: FromStr>(value: &str, expect_len: usize) -> Option<T> {
     match value.parse() {
-        Ok(result) if value.len() == expect_len => Ok(result),
-        _ => Err(Error::invalid_log_path(location)),
```

Comment:
Do we still need the `invalid_log_path` variant? If not, let's go ahead and remove it. Otherwise, disregard.